### DISCUSSION TITLE
Trying to install Chatwoot with docker

### DISCUSSION THREAD

**Discussion by Toninobandolero** [chatwoot/chatwoot#3786] [Self hosted] [answered]
Hello and thanks for this great APP.

I am trying to install it with docker and docker-compose on a vps running ubuntu 20.04.3 LTS.

First of all, it is not clear to me which guide I should follow, the Docker Chatwoot Production deployment guide or the Docker Setup. Bearing in mind that what I want to do is install Chatwoot together with Wordpress, a mail server and some other application. Each of them running in different containers.

In any case I have started following both guides without success: with the [docker setup guide](https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker#pre-requisites), 
the problem is PG::ConnectionBad: FATAL: password authentication failed for user "postgres". 
If I follow the [Docker Chatwoot Production deployment guide](https://www.chatwoot.com/docs/self-hosted/deployment/docker), I don't understand very well, but I only have 2 config files in my chatwoot folder. Everything seems to work, but I can't see anything on port 3000: ERR_CONNECTION_TIMED_OUT

I just need a common chatwoot install. In principle I will not make changes.

I will be very grateful for all the help you give me

**Comment by vishnu-narayanan** [comment]
Hey @Toninobandolero, you should be following the docker guide under the self-hosted section. The docker guide under the contribution section is for development.

Could you verify if everything is running via `docker ps` ?

**Reply by Toninobandolero** [reply]
Hi @vishnu-narayanan!
Thanks for responding
So i have to follow this one: https://www.chatwoot.com/docs/self-hosted/deployment/docker
How many containers have to be runnig?
I´m already running Wordpress on my hosting with nginx:1.15.12-alpine  wordpress:5.1.1-fpm-alpine and mysql:8.0 Do i need any specification with this?
Do i need to run everything under root user? or can be a sudo user?
Do i just to copy those 2 files .env and docker-compose.yaml into my chatwoot folder?
It is ok just with changing the password (any password) on .env Redis and Postgres, and Postgres (same on .env) on docker-compose.yaml?
Is there any specification for the passwords on Postgres and Redis?

**Reply by vishnu-narayanan** [reply]
>How many containers have to be runnig?

Chatwoot web, worker, redis and postgres.

>I´m already running Wordpress on my hosting with nginx:1.15.12-alpine wordpress:5.1.1-fpm-alpine and mysql:8.0 Do i need any specification with this?

Not really unless any of the ports required for your existing setup conflicts with chatwoot.

>Do i need to run everything under root user? or can be a sudo user?

Run it like how you have configured Docker. Any method should be fine as long as you have enough permission to run `docker`/

> Do i just to copy those 2 files .env and docker-compose.yaml into my chatwoot folder?

Yes.

>It is ok just with changing the password (any password) on .env Redis and Postgres, and Postgres (same on .env) 
on docker-compose.yaml?

Yes, that should be enough.

> Is there any specification for the passwords on Postgres and Redis?

Not really.

**Reply by Toninobandolero** [reply]
Thanks a lot for your response. 

I follow all the step, and  cannot access the APP through the port 3000: ERR_CONNECTION_TIMED_OUT. 
As i'm still practicing with all this stuff, i would like to ask you in another way: 

I WILL TRANSLATE THIS QUESTION TO THE DOCKER SELF HOSTED DISCUSSION

what is the best option for you to install Chatwoot with Docker, using a vps running ubuntu 20.04LTS as server, and sharing with other applications as Wordpress or Admirel as containers?
Will Nginx Proxy Manager or Traefic simplify the management with this configuration?
Should i use all these containers with only one docker-compose.yml file?

Thank you very much!

**Comment by Toninobandolero** [comment, marked as answer]
docker-compose run --rm rails bundle exec rails db:chatwoot_prepare
Creating network "chatwoot_default" with the default driver
Pulling postgres (postgres:12)...
12: Pulling from library/postgres
a2abf6c4d29d: Pull complete
e1769f49f910: Pull complete
33a59cfee47c: Pull complete
461b2090c345: Pull complete
8ed8ab6290ac: Pull complete
495e42c822a0: Pull complete
18e858c71c58: Pull complete
594792c80d5f: Pull complete
0cc21527d805: Pull complete
c33105c93a6d: Pull complete
f0d44539f7e1: Pull complete
2237b54399af: Pull complete
59525896cd85: Pull complete
Digest: sha256:1d098cd3c1a7b132edc5bfdd7d775ff0949104b150e31d52c0aff7bdcd25c53e
Status: Downloaded newer image for postgres:12
Pulling redis (redis:alpine)...
alpine: Pulling from library/redis
59bf1c3509f3: Pull complete
719adce26c52: Pull complete
b8f35e378c31: Pull complete
d034517f789c: Pull complete
3772d4d76753: Pull complete
211a7f52febb: Pull complete
Digest: sha256:4bed291aa5efb9f0d77b76ff7d4ab71eee410962965d052552db1fb80576431d
Status: Downloaded newer image for redis:alpine
Pulling rails (chatwoot/chatwoot:latest)...
latest: Pulling from chatwoot/chatwoot
97518928ae5f: Pull complete
d879f3f43643: Pull complete
f6e3e74a152d: Pull complete
ab54e33b1bb3: Pull complete
525c9e55fcc4: Pull complete
144ad8010912: Pull complete
93eeb60c431d: Pull complete
fab6dabc2113: Pull complete
6e51b34a747f: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:46352862416c5c5d2f65ba486307d87c951bf46df2787f3cf6bcac023ba0c49e
Status: Downloaded newer image for chatwoot/chatwoot:latest
Creating chatwoot_redis_1    ... done
Creating chatwoot_postgres_1 ... done
Creating chatwoot_rails_run  ... done
+ rm -rf /app/tmp/pids/server.pid
+ rm -rf '/app/tmp/cache/*'
+ echo 'Waiting for postgres to become ready....'
Waiting for postgres to become ready....
+ docker/entrypoints/helpers/pg_database_url.rb
+ export 'POSTGRES_PORT=5432'
+ PG_READY='pg_isready -h postgres -p 5432 -U postgres'
+ pg_isready -h postgres -p 5432 -U postgres
postgres:5432 - accepting connections
+ echo 'Database ready to accept connections.'
Database ready to accept connections.
+ bundle install
Using rake 13.0.6
Using concurrent-ruby 1.1.9
Using i18n 1.8.11
Using minitest 5.15.0
Using tzinfo 2.0.4
Using zeitwerk 2.5.1
Using activesupport 6.1.4.3
Using builder 3.2.4
Using erubi 1.10.0
Using mini_portile2 2.5.3
Using racc 1.6.0
Using nokogiri 1.11.7
Using rails-dom-testing 2.0.3
Using crass 1.0.6
Using loofah 2.13.0
Using rails-html-sanitizer 1.4.2
Using actionview 6.1.4.3
Using rack 2.2.3
Using rack-test 1.1.0
Using actionpack 6.1.4.3
Using nio4r 2.5.8
Using websocket-extensions 0.1.5
Using websocket-driver 0.7.5
Using actioncable 6.1.4.3
Using globalid 1.0.0
Using activejob 6.1.4.3
Using activemodel 6.1.4.3
Using activerecord 6.1.4.3
Using marcel 1.0.2
Using mini_mime 1.1.2
Using activestorage 6.1.4.3
Using mail 2.7.1
Using actionmailbox 6.1.4.3
Using actionmailer 6.1.4.3
Using actiontext 6.1.4.3
Using activerecord-import 1.2.0
Using acts-as-taggable-on 8.1.0
Using public_suffix 4.0.6
Using addressable 2.8.0
Using method_source 1.0.0
Using thor 1.1.0
Using railties 6.1.4.3
Using momentjs-rails 2.20.1
Using datetime_picker_rails 0.0.7
Using jquery-rails 4.4.0
Using kaminari-core 1.2.1
Using kaminari-actionview 1.2.1
Using kaminari-activerecord 1.2.1
Using kaminari 1.2.1
Using ffi 1.15.4
Using sassc 2.4.0
Using sprockets 4.0.2
Using sprockets-rails 3.4.2
Using tilt 2.0.10
Using sassc-rails 2.1.2
Using selectize-rails 0.12.6
Using administrate 0.16.0
Using ast 2.4.2
Using attr_extras 6.2.4
Using aws-eventstream 1.2.0
Using aws-partitions 1.513.0
Using aws-sigv4 1.4.0
Using jmespath 1.4.0
Using aws-sdk-core 3.121.1
Using aws-sdk-kms 1.49.0
Using aws-sdk-s3 1.103.0
Using multipart-post 2.1.1
Using faraday 1.0.1
Using faraday_middleware 1.0.0
Using connection_pool 2.2.5
Using net-http-persistent 4.0.1
Using azure-storage-common 2.0.2
Using azure-storage-blob 2.0.1
Using multi_json 1.15.0
Using statsd-ruby 1.5.0
Using barnes 0.0.9
Using bcrypt 3.1.16
Using msgpack 1.4.2
Using bootsnap 1.9.1
Using brakeman 5.1.1
Using browser 5.3.1
Using bundler 2.2.33
Using commonmarker 0.23.2
Using ddtrace 0.53.0
Using declarative 0.0.20
Using orm_adapter 0.5.0
Using responders 3.0.1
Using warden 1.2.9
Using devise 4.8.0
Using devise-secure_password 2.0.1 from https://github.com/chatwoot/devise-secure_password
Using rails 6.1.4.3
Using devise_token_auth 1.2.0
Using digest-crc 0.6.4
Using unf_ext 0.0.8
Using unf 0.1.4
Using domain_name 0.5.20190701
Using dotenv 2.7.6
Using dotenv-rails 2.7.6
Using down 5.2.4
Using regexp_parser 2.1.1
Using ecma-re-validator 0.3.0
Using email_reply_trimmer 0.1.13
Using et-orbi 1.2.5
Using execjs 2.8.1
Using mime-types-data 3.2021.0901
Using mime-types 3.3.1
Using multi_xml 0.6.0
Using httparty 0.20.0
Using facebook-messenger 2.0.1
Using fcm 1.0.3
Using flag_shih_tzu 0.3.23
Using foreman 0.87.2
Using raabro 1.4.0
Using fugit 1.5.2
Using google-protobuf 3.19.2
Using googleapis-common-protos-types 1.2.0
Using grpc 1.41.0
Using googleapis-common-protos 1.3.12
Using jwt 2.3.0
Using memoist 0.16.2
Using os 1.1.1
Using signet 0.16.0
Using googleauth 0.17.1
Using gapic-common 0.3.4
Using geocoder 1.7.0
Using gli 2.20.1
Using httpclient 2.8.3
Using trailblazer-option 0.1.1
Using uber 0.1.0
Using representable 3.1.1
Using retriable 3.1.2
Using rexml 3.2.5
Using webrick 1.7.0
Using google-apis-core 0.4.1
Using google-apis-iamcredentials_v1 0.7.0
Using google-apis-storage_v1 0.8.0
Using google-cloud-env 1.5.0
Using google-cloud-errors 1.2.0
Using google-cloud-core 1.6.0
Using google-cloud-dialogflow-v2 0.6.4
Using google-cloud-dialogflow 1.2.0
Using google-cloud-storage 1.34.1
Using groupdate 5.2.2
Using haikunator 1.1.1
Using sexp_processor 4.15.3
Using ruby_parser 3.17.0
Using ruby2ruby 2.4.4
Using hairtrigger 0.2.24
Using hana 1.3.7
Using hashie 4.1.0
Using hkdf 0.3.0
Using html2text 0.2.1
Using http-accept 1.7.0
Using http-cookie 1.0.4
Using mini_magick 4.11.0
Using ruby-vips 2.1.3
Using image_processing 1.12.1
Using jbuilder 2.11.2
Using json 2.5.1
Using uri_template 0.7.0
Using json_schemer 0.2.18
Using koala 3.0.0
Using line-bot-api 1.22.0
Using liquid 5.1.0
Using maxminddb 0.1.22
Using netrc 0.11.0
Using newrelic_rpm 8.0.0
Using oauth 0.5.6
Using parser 3.0.2.0
Using pg 1.2.3
Using procore-sift 0.16.0
Using puma 5.5.2
Using pundit 2.1.1
Using rack-attack 6.5.0
Using rack-cors 1.1.1
Using rack-proxy 0.7.0
Using rack-timeout 0.6.0
Using redis 4.4.0
Using redis-namespace 1.8.1
Using rest-client 2.1.0
Using scout_apm 4.1.2
Using semantic_range 3.0.0
Using sentry-ruby-core 4.7.3
Using sentry-rails 4.7.3
Using sentry-ruby 4.7.3
Using sidekiq 6.2.2
Using sentry-sidekiq 4.7.3
Using sidekiq-cron 1.2.0
Using slack-ruby-client 0.17.0
Using telephone_number 1.4.12
Using time_diff 0.3.0
Using twilio-ruby 5.32.0
Using twitty 0.1.4
Using tzinfo-data 1.2021.3
Using uglifier 4.2.0
Using valid_email2 4.0.0
Using webpacker 5.4.3
Using webpush 1.1.0
Using wisper 2.0.0
Bundle complete! 99 Gemfile dependencies, 199 gems now installed.
Gems in the groups 'development' and 'test' were not installed.
Bundled gems are installed into `/gems`
+ BUNDLE='bundle check'
+ bundle check
The Gemfile's dependencies are satisfied
+ exec bundle exec rails db:chatwoot_prepare
Rspec gem is missing. Install it.
Flay gem is missing. Install it.
Rspec gem is missing. Install it.
rails aborted!
ActiveRecord::ConnectionNotEstablished: FATAL:  password authentication failed for user "postgres"
/app/lib/tasks/db_enhancements.rake:21:in `block (3 levels) in <main>'
/app/lib/tasks/db_enhancements.rake:18:in `each'
/app/lib/tasks/db_enhancements.rake:18:in `block (2 levels) in <main>'

Caused by:
PG::ConnectionBad: FATAL:  password authentication failed for user "postgres"
/app/lib/tasks/db_enhancements.rake:21:in `block (3 levels) in <main>'
/app/lib/tasks/db_enhancements.rake:18:in `each'
/app/lib/tasks/db_enhancements.rake:18:in `block (2 levels) in <main>'
Tasks: TOP => db:chatwoot_prepare
(See full trace by running task with --trace)
ERROR: 1

**Reply by vishnu-narayanan** [reply]
Looks like you are having issues with postgres authentication. I am assuming you changed passwords midway Could you try clearing everything(dokcer volumes and all) and try fresh?

**Reply by vishnu-narayanan** [reply]
https://github.com/chatwoot/chatwoot/discussions/3323

**Reply by vishnu-narayanan** [reply]
refer to the above link to remove your postgres data and try again?

**Comment by Toninobandolero** [comment]
docker-compose up -d
chatwoot_postgres_1 is up-to-date
Creating chatwoot_base_1 ... 
Creating chatwoot_base_1    ... done
Creating chatwoot_rails_1   ... done
Creating chatwoot_sidekiq_1 ... done

**Comment by Toninobandolero** [comment]
docker ps

"bundle exec sidekiq…"   44 seconds ago   Up 42 seconds   3000/tcp                                   chatwoot_sidekiq_1
"docker/entrypoints/…"   44 seconds ago   Up 42 seconds   127.0.0.1:3000->3000/tcp                   chatwoot_rails_1
"docker-entrypoint.s…"   2 minutes ago    Up 2 minutes    127.0.0.1:6379->6379/tcp                   chatwoot_redis_1
 "docker-entrypoint.s…"   2 minutes ago    Up 2 minutes    127.0.0.1:5432->5432/tcp                   chatwoot_postgres_1
